### PR TITLE
Zen 458 group booking disable

### PR DESCRIPTION
**Pull request by daniel112**
**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-458)

## Context

* in group booking mode, when every single person is booked to a session. Any adjacent session should have its booking button disabled. Just like the single booking mode

## Goal

* group booking conflict - unlimited is an edge case that was missed
* exctract out `getDisabled` method so it can be unit tested

## Updates

* `src/mocks/eventsforce/bookingStates/conflict.js`
    * updated conflict test data
* `src/utils/models/sessions/bookingStateFactory.js`
    * extracted `getDisabled` to its own file
* `src/utils/models/sessions/getDisabled.js`
    * the extracted method from `bookingStateFactory`
    *  minor adjust so it will disable when capacity is unlimited && conflict exists
### Unit tests
* run `keg evf pack run package=docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-458-group-booking-disable command=test`
* verify all test passes

* run `keg evf pack run package=docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-458-group-booking-disable`
* navigate to http://evf-zen-458-group-booking-disable.local.kegdev.xyz/

### Single booking conflicts
* verify this still works the same as before
* update the JSON booking data to `conflict-single-limited`
* verify if you book a session that has the same day and time as another session, that session booking btn is disabled
* repeat with JSON booking data `conflict-single-unlimited`

### Group booking conflicts
on these booking data: (`conflict-group-limited` && `conflict-group-unlimited`)
* verify if you book all the possible attendees in the group to Session A. Session B's booking button  will be disabled. and vice versa
    * [image](https://user-images.githubusercontent.com/3317835/102927484-53df0b00-4454-11eb-86d6-b95122640e31.png)

### Spot Check
* Modify the JSON data for other disabled button variation (single && group booking)
    * `fully booked` state should disable the button
    * modifying `session.capacity.remainingPlaces === 0` && `session.capacity.isWaitingListAvailable === false` should disable the button

### CODE REVIEW

File: src/utils/models/sessions/getDisabled.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * TODO: Sync with events force to have them pass an option for bookingStopped
+ * Will need to update this method to pull that option based on how its passed in
+ */
+/**
+ * Checks if the booking display should be disabled from interaction
+ * @param {import('SVModels/session').Session} props.session
+ *
+ * @returns {boolean} - If booking changes have been stopped for this session
+ */
+const getBookingStopped = session => {
```
Comment (lancetipton) [inline src/utils/models/sessions/getDisabled.js:15]: This can actually be removed. According to Event-Force, the edge case this was covering will never exist. If it does, it's something they should fix on their end. 

So can you just remove this. Thanks

File: src/mocks/eventsforce/bookingStates/conflict.js [hunk 2]
```diff
@@ -1,59 +1,60 @@
...
-        remainingPlaces: 5,
-      },
-      price: {
-        currency: 'USD',
-        amount: 923.0,
-      },
+const sessions = (isUnlimited = false) => [
+  {
+    allowBooking: true,
+    // Custom identifier to flag that booking has been stopped for that session
```
Comment (lancetipton) [inline src/mocks/eventsforce/bookingStates/conflict.js:6]: See my other comment, but you can just remove this flag.

**Review by lancetipton — APPROVED** [review]
Changes look good. Approving because everything looks good, but left a comment about code that can be removed.

File: src/utils/models/sessions/getDisabled.js (hunk 1 above)
Comment (daniel112) [reply src/utils/models/sessions/getDisabled.js:15]: will do

File: src/mocks/eventsforce/bookingStates/conflict.js (hunk 2 above)
Comment (daniel112) [reply src/mocks/eventsforce/bookingStates/conflict.js]: done
